netdeploy/network.go

```diff
@@ -337,21 +335,38 @@ func (n Network) GetPeerAddresses(binDir string) []string {
 		if err != nil {
 			continue
 		}
-		if strings.HasPrefix(relayAddress, "http://") {
-			relayAddress = relayAddress[7:]
-		}
-		peerAddresses = append(peerAddresses, relayAddress)
+		peerAddresses = append(peerAddresses, strings.TrimPrefix(relayAddress, "http://"))
 	}
 	return peerAddresses
 }
 
-func (n Network) startNodes(binDir, relayAddress string, redirectOutput bool) error {
-	args := nodecontrol.AlgodStartArgs{
-		PeerAddress:       relayAddress,
-		RedirectOutput:    redirectOutput,
-		ExitErrorCallback: n.nodeExitCallback,
+func (n Network) startNodes(binDir string, relayNameToAddress map[string]string, redirectOutput bool) error {
+	allRelaysAddresses := strings.Join(maps.Values(relayNameToAddress), ";")
+
+	nodeConfigToEntry := make(map[string]remote.NodeConfigGoal, len(n.cfg.Template.Nodes))
+	for _, n := range n.cfg.Template.Nodes {
+		nodeConfigToEntry[n.Name] = n
 	}
+
 	for _, nodeDir := range n.nodeDirs {
+		args := nodecontrol.AlgodStartArgs{
+			PeerAddress:       allRelaysAddresses,
+			RedirectOutput:    redirectOutput,
+			ExitErrorCallback: n.nodeExitCallback,
+		}
+		if n, ok := nodeConfigToEntry[nodeDir]; ok && len(n.PeerList) > 0 {
+			relayNames := strings.Split(n.PeerList, ";")
```

Comment:
Why is the PeerList constructed as a joined string instead of handled as a natural `[]string`, I am assuming there's a serialization concern? I just notice that the nodeConfig object already uses `[]NodeWalletData` so I figured it could support a slice of addresses

Reply:
I decided to use the same syntax as `-p` / phonebook for consistency, and it looks less typing. But I do not have a strong opinion there